pkg/codegen/nodejs/gen.go

```diff
@@ -717,6 +793,47 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) error {
 	argsComment := fmt.Sprintf("The set of arguments for constructing a %s resource.", name)
 	mod.genPlainType(w, argsType, argsComment, r.InputProperties, true, false, 0)
 
+	// Emit any method types inside a namespace merged with the class, to represent types nested in the class.
+	// https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-namespaces-with-classes
+	genMethodTypes := func(w io.Writer, method *schema.Method) {
+		fun := method.Function
+		methodName := title(method.Name)
+		if fun.Inputs != nil {
+			args := make([]*schema.Property, 0, len(fun.Inputs.InputShape.Properties))
+			for _, arg := range fun.Inputs.InputShape.Properties {
+				if arg.Name == "__self__" {
+					continue
+				}
+				args = append(args, arg)
+			}
+			if len(args) > 0 {
+				comment := fun.Inputs.Comment
+				if comment == "" {
+					comment = fmt.Sprintf("The set of arguments for the %s.%s method.", name, method.Name)
+				}
+				mod.genPlainType(w, methodName+"Args", comment, args, true, false, 1)
```

Comment:
When working on codegen, these positional booleans were making it harder for me to read and keep track of what's this doing. Don't fix in this PR but consider maybe we should name them in a helper record or something so call  sites are easier to read.